### PR TITLE
handle cities with spaces in the urllib.request version of weather05.py

### LINKED ISSUE #3870: Weather05.py doesn't handle cities with spaces
Weather05.py uses the urllib.Request library, which doesn't automatically urlencode.  It crashes when presented with a city name such as `New York`.  I'm submitting a PR to fix this.

### PR DESCRIPTION
**Pull request by danschwarz**
**Please review the following checklist.**

- [ ] Docstrings on all new or modified functions / classes 
- [ ] Updated documentation
- [ ] Updated CHANGELOG.md (where appropriate)

None of the above checklist items apply (this is probably too minor to make the changelog).

### CODE REVIEW

**Comment by danschwarz** [comment]
fixes #3870

**Review by rodrigogiraoserrao — APPROVED** [review]
Well spotted & thanks for the fix.

File: docs/examples/guide/workers/weather05.py [hunk 1]
```diff
@@ -21,7 +22,7 @@ def compose(self) -> ComposeResult:
 
     async def on_input_changed(self, message: Input.Changed) -> None:
         """Called when the input changes"""
-        self.update_weather(message.value)
+        self.update_weather(quote(message.value))
```
Comment (willmcgugan) [inline docs/examples/guide/workers/weather05.py:25]: I think a better place for the `quote` would be the `update_weather` method itself.

Comment (rodrigogiraoserrao) [reply docs/examples/guide/workers/weather05.py:25]: Even better spotted 🚀

Comment (rodrigogiraoserrao) [reply docs/examples/guide/workers/weather05.py]: @danschwarz I went ahead and moved the `quote` for you so we can deploy the docs fix ASAP!
I hope you don't mind 😄

**Comment by willmcgugan** [comment]
@rodrigogiraoserrao Bear in mind that weather05.py is used in the docs. If the lines numbers have changed, you will need to update `hl_lines` in the appropriate Markdown file.

**Comment by rodrigogiraoserrao** [comment]
> @rodrigogiraoserrao Bear in mind that weather05.py is used in the docs. If the lines numbers have changed, you will need to update `hl_lines` in the appropriate Markdown file.

True. See #3875.

**Pull request merged**